esp/clients/roxiecontrol.cpp

```diff
@@ -18,12 +18,81 @@
 #include "roxiecontrol.hpp"
 #include "jmisc.hpp"
 #include "securesocket.hpp"
+#include "jsecrets.hpp"
 
 const unsigned roxieQueryRoxieTimeOut = 60000;
 
 #define EMPTY_RESULT_FAILURE 1200
 #define SECURE_CONNECTION_FAILURE 1201
 
+CConnectionSettings::CConnectionSettings(IPropertyTree &service)
+{
+    name.set(service.queryProp("@name"));
+    if (isEmptyString(name))
+        throw makeStringException(-1, "CConnectionSettings: name required for service configuration");
+
+    port = service.getPropInt("@port");
+    if (port == 0)
+        throw makeStringException(-1, "CConnectionSettings: invalid port for service configuration");
+
+    tlsService  = service.getPropBool("@tls");
+    if (tlsService)
+        tlsConfig.setown(createTlsClientSecretInfo(service.queryProp("@issuer"), !service.getPropBool("@public"), service.getPropBool("@selfSigned"), service.getPropBool("@caCert")));
+
+    readFQDNs();
+}
+
+void CConnectionSettings::readFQDNs()
+{
+    char *targets = (char *) name.get();
+    char *dash = strrchr(targets, '-');
+    if (dash)
+    {
+        parseIPRange(targets, dash);
+        shuffleFQDNs();
+    }
+    else
+    {
+        FQDNs.append(name);
+    }
+}
+
+void CConnectionSettings::parseIPRange(char *targets, char *dash)
+{
+    *dash = 0;
+    int numberAfterDash = atoi(dash+1);
+    char *lastDot = strrchr(targets, '.');
+    *lastDot = 0;
+    int numberBeforeDash = atoi(lastDot+1);
+    for (unsigned i = numberBeforeDash; i <= numberAfterDash; i++)
+    {
+        StringBuffer t;
+        t.append(targets).append('.').append(i);
+        FQDNs.append(t.str());
+    }
+}
+
+void CConnectionSettings::shuffleFQDNs()
+{
+    Owned<IRandomNumberGenerator> random = createRandomNumberGenerator();
+    random->seed((unsigned)get_cycles_now());
+
+    unsigned i = FQDNs.ordinality();
+    while (i > 1)
+    {
+        unsigned j = random->next() % i;
+        i--;
+        FQDNs.swap(i, j);
+    }
+}
+
+const char *CConnectionSettings::nextFQDN()
+{
+    const char *FQDN = FQDNs.item(nextFQDNIndex);
+    ++nextFQDNIndex %= FQDNs.ordinality();
```

Comment:
related to above - I don't see how this fits in with the schema of a roxie service definition.